pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -1151,18 +1196,46 @@ func TestSidecarOutboundHTTPRouteConfig(t *testing.T) {
 				"test.com:8080": {
 					"test.com:8080": true, "8.8.8.8:8080": true,
 				},
+				"service-A:7777": {
+					"service-A:7777": true,
+				},
 				"block_all": {
 					"*": true,
 				},
 			},
 			registryOnly: true,
 		},
+		{
+			name:                  "virtual service hosts with subsets and with existing service",
+			routeName:             "7777",
+			sidecarConfig:         sidecarConfigWithAllowAny,
+			virtualServiceConfigs: []*config.Config{&virtualService7},
+			expectedHosts: map[string]map[string]bool{
+				"allow_any": {
+					"*": true,
+				},
+				"service-A:7777": {
+					"service-A":      true,
+					"service-A:7777": true,
+				},
+				"service-A.v2:7777": {
+					"service-A.v2":      true,
+					"service-A.v2:7777": true,
+				},
+				"service-A.v3:7777": {
+					"service-A.v3":      true,
+					"service-A.v3:7777": true,
+				},
+			},
+			expectedRoutes: 7,
```

Comment:
trying to understand why it is 7

Reply:
Because of the header match route in VS it creates multiple routes

```
{name:"service-A.v2:7777"  domains:"service-A.v2"  domains:"service-A.v2:7777"  routes:{match:{prefix:"/"  case_sensitive:{value:true}  headers:{name:":authority"  exact_match:"service-A.v2"}}  route:{cluster:"outbound|7777|v2|service-A"  timeout:{}  retry_policy:{retry_on:"connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes"  num_retries:{value:2}  retry_host_predicate:{name:"envoy.retry_host_predicates.previous_hosts"}  host_selection_retry_max_attempts:5  retriable_status_codes:503}  max_grpc_timeout:{}}  metadata:{filter_metadata:{key:"istio"  value:{fields:{key:"config"  value:{string_value:"/apis/networking.istio.io/v1alpha3/namespaces//virtual-service/vs-1"}}}}}  decorator:{operation:"service-A:7777/*"}}  routes:{match:{prefix:"/"  case_sensitive:{value:true}  headers:{name:":authority"  exact_match:"service-A.v3"}}  route:{cluster:"outbound|7777|v3|service-A"  timeout:{}  retry_policy:{retry_on:"connect-failure,refused-stream,unavailable,cancelled,ret
riable-status-codes"  num_retries:{value:2}  retry_host_predicate:{name:"envoy.retry_host_predicates.previous_hosts"}  host_selection_retry_max_attempts:5  retriable_status_codes:503}  max_grpc_timeout:{}}  metadata:{filter_metadata:{key:"istio"  value:{fields:{key:"config"  value:{string_value:"/apis/networking.istio.io/v1alpha3/namespaces//virtual-service/vs-1"}}}}}  decorator:{operation:"service-A:7777/*"}}  include_request_attempt_count:true}
```

Reply:
@hzxuzhonghu are you good with this?

Reply:
Understood